doozerlib/distgit.py

```diff
@@ -1762,18 +1762,25 @@ def _update_yum_update_commands(self, force_yum_updates: bool, df_fileobj: io.Te
             if not force_yum_updates or not line.startswith('FROM '):
                 continue
             build_stage += 1
-            # If the curent user inherited from the base image for this stage is not root, `yum update -y` will fail.
-            # We need to change the current user to root.
-            # For non-final stages, it should be safe to inject "USER 0" before `yum update -y`.
-            # However for the final stage, injecting "USER 0" without changing the user back may cause unexpected behavior.
-            # Per https://github.com/openshift/doozer/pull/428#issuecomment-861795424, introduce a new metadata `final_stage_user` for images so we can switch the user back later.
-            if build_stage < build_stage_num or final_stage_user:
+
+            if build_stage != build_stage_num:
+                # If this is not the final stage, ignore this FROM
+                continue
```

Comment:
[DockerfileParser.add_lines](https://github.com/containerbuildsystem/dockerfile-parse/blob/9f22a64d8cd037cb191aca69dc36bba23a831aab/dockerfile_parse/parser.py#L735) ought to simplify this. Also can't we remove the bit above checking for rhel7?

Reply:
We still need `yum install -y yum-plugin-priorities` for RHEL7, but the `if` check doesn't seem to be needed anymore.